src/HackF5.UnitySpy/AssemblyImageFactory.cs

```diff
@@ -42,29 +42,48 @@ public static IAssemblyImage Create(int processId, string assemblyName = "Assemb
+        private static AssemblyImage GetAssemblyImage(ProcessFacade process, string name, IntPtr rootDomainFunctionAddress)
         {
-            var domainAddress = process.ReadPtr((uint)rootDomainFunctionAddress + 1);
-            //// pointer to struct of type _MonoDomain
-            var domain = process.ReadPtr(domainAddress);
+
+            IntPtr domain;
+            if (process.Is64Bits)
+            {
+                // Offsets taken by decompiling the 64 bits version of mono-2.0-bdwgc.dll
+                // mov rax, [rip + 0x46ad39]
+                // ret
+                var offset = process.ReadInt32(rootDomainFunctionAddress + 3) + 7;
+                //// pointer to struct of type _MonoDomain
+                domain = process.ReadPtr(rootDomainFunctionAddress + offset);
+            } 
+            else
+            {
+                var domainAddress = process.ReadPtr(rootDomainFunctionAddress + 1);
+                //// pointer to struct of type _MonoDomain
+                domain = process.ReadPtr(domainAddress);
+            }
 
             //// pointer to array of structs of type _MonoAssembly
-            var assemblyArrayAddress = process.ReadPtr(domain + MonoLibraryOffsets.ReferencedAssemblies);
+            var assemblyArrayAddress = process.ReadPtr(domain + process.MonoLibraryOffsets.ReferencedAssemblies);
             for (var assemblyAddress = assemblyArrayAddress;
                 assemblyAddress != Constants.NullPtr;
-                assemblyAddress = process.ReadPtr(assemblyAddress + 0x4))
+                assemblyAddress = process.ReadPtr(assemblyAddress + process.SizeOfPtr))
             {
                 var assembly = process.ReadPtr(assemblyAddress);
-                var assemblyNameAddress = process.ReadPtr(assembly + 0x8);
+                var assemblyNameAddress = process.ReadPtr(assembly + (process.SizeOfPtr * 2));
                 var assemblyName = process.ReadAsciiString(assemblyNameAddress);
                 if (assemblyName == name)
                 {
-                    return new AssemblyImage(process, process.ReadPtr(assembly + MonoLibraryOffsets.AssemblyImage));
+                    return new AssemblyImage(process, process.ReadPtr(assembly + process.MonoLibraryOffsets.AssemblyImage));
+                    //return new AssemblyImage(process, process.ReadPtr(assembly + 0x44)); -> CollectionManager is null if we have this
+                    //return new AssemblyImage(process, process.ReadPtr(assembly + 0x54)); -> CollectionManager is null if we have this
+                    //return new AssemblyImage(process, process.ReadPtr(assembly + 0x6c)); -> CollectionManager is null if we have this
+                    //return new AssemblyImage(process, process.ReadPtr(assembly + 0xac)); -> CollectionManager is null if we have this
+                    //return new AssemblyImage(process, process.ReadPtr(assembly + 0xac));
```

Comment:
Do these commented lines help understand better the code? If so, maybe the comments could be moved to the `process.MonoLibraryOffsets.AssemblyImage` field? Otherwise maybe just remove them?

Reply:
Edit: I just realized that these come from the other UnitySpy fork, so I think you can safely remove them completely